securetar/__init__.py

```diff
@@ -143,16 +145,15 @@ def __exit__(self, exc_type, exc_value, traceback) -> None:
             self._tar.close()
             self._tar = None
         if self._file:
+            if not self._mode.startswith("r"):
+                self._file.write(self._encrypt.update(self._padder.finalize()))
             if not self._fileobj:
                 self._file.close()
             self._file = None
 
     def write(self, data: bytes) -> None:
         """Write data."""
-        if len(data) % BLOCK_SIZE != 0:
-            padder = padding.PKCS7(BLOCK_SIZE_BITS).padder()
-            data = padder.update(data) + padder.finalize()
-
+        data = self._padder.update(data)
```

Comment:
Note that this doesn't do any operation on the buffer passed in, except for keeping track of the total length to be able to write PKCS7 compliant padding in the end.